src/common/types/metadata.js

```diff
@@ -108,13 +108,38 @@ type MetadataEditor = {
     template: MetadataTemplate,
 };
 
+type MetadataInstanceTemplateField = {
+    description?: string,
+    displayName: string,
+    hidden?: boolean,
+    id: string,
+    isHidden?: boolean,
+    key: string, // V2
+    options?: Array<MetadataTemplateFieldOption>, // V3
+    type: MetadataFieldType,
+    value: string,
+};
+
+type MetadataInstanceTemplateFields = { [string]: MetadataInstanceTemplateField };
```

Comment:
Why this additional identifier is needed? Could we use array of `MetadataInstanceTemplateField` instead?

Reply:
This is actually good point :) I'll change that